### PR TITLE
Improve docs and example README for MLflow Docker projects

### PR DESCRIPTION
**Pull request by dbczumar**
This PR improves the MLflow documentation and example README for the Docker projects feature.

### CODE REVIEW

File: docs/source/projects.rst [hunk 1]
```diff
@@ -56,24 +55,106 @@ project for remote execution on `Databricks <https://databricks.com>`_.
...
+Docker Containers
+  You can also run MLflow Projects inside 
+  `Docker containers <https://www.docker.com/resources/what-container>`_, which allows you to 
+  capture non-Python dependencies such as Java libraries. When you run an MLflow Project that 
+  specifies a Docker image, MLflow executes the image, mounts the project directory in the resulting 
+  container at ``/mlflow/projects/code``, and invokes the project entry point in the container. 
+ 
+  Environment variables, such as ``MLFLOW_TRACKING_URI``, are propagated inside the Docker container 
+  during project execution. Additionally, :ref:`Runs <concepts>` and 
+  :ref:`Experiments <organizing-runs-in-experiments>` created by the project are saved to the
```
Comment (stbof) [inline docs/source/projects.rst]: Both runs and experiments should be lowercase; they are not proper nouns.

File: docs/source/projects.rst [hunk 2]
```diff
@@ -56,24 +55,106 @@ project for remote execution on `Databricks <https://databricks.com>`_.
...
+  capture non-Python dependencies such as Java libraries. When you run an MLflow Project that 
+  specifies a Docker image, MLflow executes the image, mounts the project directory in the resulting 
+  container at ``/mlflow/projects/code``, and invokes the project entry point in the container. 
+ 
+  Environment variables, such as ``MLFLOW_TRACKING_URI``, are propagated inside the Docker container 
+  during project execution. Additionally, :ref:`Runs <concepts>` and 
+  :ref:`Experiments <organizing-runs-in-experiments>` created by the project are saved to the 
+  tracking server specified by your :ref:`tracking URI <where-runs-are-recorded>`. When running 
+  against a local tracking URI, MLflow will mount the host system's tracking directory
+  (e.g., a local ``mlruns`` directory) inside the container so that metrics, parameters, and
```
Comment (stbof) [inline docs/source/projects.rst]: MLflow will mount -> MLflow mounts

File: docs/source/projects.rst [hunk 3]
```diff
@@ -56,24 +55,106 @@ project for remote execution on `Databricks <https://databricks.com>`_.
...
+  file in the root of the project directory, or by including a ``conda_env`` entry in your
+  ``MLProject`` file. For more information, see the :ref:`project-directories` and 
+  :ref:`MLProject File <mlproject-file>` sections.
+
+.. _project-docker-container-environments:
+
+Docker container
+  `Docker containers <https://www.docker.com/resources/what-container>`_ allow you to 
+  capture non-Python dependencies such as Java libraries. When you run an MLflow Project that 
+  specifies a Docker image, MLflow runs the image, mounts the project directory in the resulting
```
Comment (smurching) [inline docs/source/projects.rst]: We actually build a new image based on the project-specified image with the project directory COPY'd in to `/mlflow/projects/code`, so maybe we can reword to something like:

```
When you run an MLflow Project that specifies a Docker image IMG, MLflow builds a new image based on IMG with the project directory copied into `/mlflow/projects/code`. It then runs the image, invoking the project entry point in the resulting container.
```

**Review by smurching — COMMENTED** [review]
This is great, one comment but otherwise LGTM, thanks @dbczumar!

File: docs/source/projects.rst (hunk 3 above)
Comment (dbczumar) [reply docs/source/projects.rst]: @smurching Thanks for the clarification! I've updated the phrasing to indicating that the directory gets copied. Let me know if this looks okay!

**Comment by stbof** [comment]
LGTM.

**Review by smurching — APPROVED** [review]
LGTM thanks for the updates!

**Pull request merged**